ddp-workspace/projects/ddp-dsm-ui/src/app/field-datepicker/field-datepicker.component.ts

```diff
@@ -84,88 +97,109 @@ export class FieldDatepickerComponent implements OnInit, OnChanges {
 
   public check(): void {
     this.colorDuringPatch = true;
-    if (this._dateString != null) {
-      if (this._dateString === '') {
-        this.emitDate('');
-      } else {
-        if (this._dateString !== 'N/A' && this._dateString !== 'Not Found') {
-          const tmp = Utils.parseDate(this._dateString, this.dateFormat, this.allowUnknownDay);
-          // console.log(tmp);
-          if (tmp instanceof Date) {
-            this.datePicker = tmp;
-            if (this.datePicker != null) {
-              this.emitDate(Utils.getDateFormatted(this.datePicker, Utils.DATE_STRING));
-              this.colorDuringPatch = false;
-              this.error = null;
-            } else {
-              this.error = this.dateFormat;
-            }
-          } else if (tmp === Utils.DATE_PARTIAL) {
-            this.error = null;
-            this.emitDate(Utils.getPartialDateFormatted(this._dateString, this.dateFormat));
+    if (this._dateString === '') {
+      this.emitDate('');
+    } else {
+      if (this._dateString !== 'N/A' && this._dateString !== 'Not Found') {
+        const tmp = Utils.parseDate(this._dateString, this.dateFormat, this.allowUnknownDay);
+        // console.log(tmp);
+        if (tmp instanceof Date) {
+          this.hasDateChanged = (this.datePicker !== tmp);
+          this.datePicker = tmp;
+          if (this.datePicker != null) {
+            this.emitDate(Utils.getDateFormatted(this.datePicker, Utils.DATE_STRING));
             this.colorDuringPatch = false;
-          } else if (tmp == null) {
+            this.error = null;
+          } else {
             this.error = this.dateFormat;
           }
+        } else if (tmp === Utils.DATE_PARTIAL) {
+          this.error = null;
+          this.emitDate(Utils.getPartialDateFormatted(this._dateString, this.dateFormat));
+          this.colorDuringPatch = false;
+          this.hasDateChanged = false;
+        } else if (tmp == null) {
+          this.error = this.dateFormat;
         }
       }
-    } else {
-      this.colorDuringPatch = false;
-      this.error = null;
     }
   }
 
-  public selectDate(event: any): void {
-    this.showDatePicker = false;
-    this.datePicker = event;
-    if (this.datePicker instanceof Date) {
+  public saveDate(): void{
+    this.saveButtonText = "Saving..."
+    this.currentlySaving = true;
+      this.dateSaved.subscribe({
+        next: data => {
+          this.saveButtonText = "Success!";
+          this.hasDateChanged = false;
+        },
+        error: err => {
+          this.saveButtonText = "Failed";
+        },
+          }).add(() => {
```

Comment:
**# Optional**
`.add()` function on subscription is called then when we want to add more subscriptions to it, in order to easily unsubscribe them later on. 

I'm just wondering, could we not wrap this block of code in `finalize()` [operator](https://rxjs.dev/api/operators/finalize)?
Like so:
```
 this.dateSaved
        .pipe(finalize(() => {
          this.saveCompleted.emit();
          setTimeout(() => {
            this.saveButtonText = "Save Date";
            this.currentlySaving = false;
          },5000);
        }))
        .subscribe({ ... })
```

Please read this [discussion](https://github.com/ReactiveX/rxjs/issues/2769).